client/src/components/FormAction/FormAction.js

```diff
@@ -113,6 +136,26 @@ class FormAction extends SilverStripeComponent {
   }
 
   /**
+   * Get saved title
+   *
+   * @return {String}|null
+   */
+  getSavedTitle() {
+    // In case this is specified directly
+    return this.props.savedTitle || this.props.data.savedTitle || null;
+  }
+
+  getSavedIcon() {
+    // In case this is specified directly
+    return this.props.savedIcon || this.props.data.savedIcon || null;
```

Comment:
Why are all these fields in two places? They should either be on this.props or this.props.data.